src/bundles/files/actions.js

```diff
@@ -398,6 +398,37 @@ const actions = () => ({
     }
   }),
 
+  /**
+   * Adds CAR file. On completion will trigger `doFilesFetch` to update the state.
+   * @param {FileStream} carFile
+   * @param {string} name
+   */
+  doAddCarFile: (carFile, name = '') => perform(ACTIONS.ADD_CAR_FILE, async (ipfs, { store }) => {
+    ensureMFS(store)
+
+    const stream = carFile.content.stream()
+    try {
+      const result = await all(ipfs.dag.import(stream, {
+        pinRoots: true
+      }))
+      const cid = result[0].root.cid
+      const src = `/ipfs/${cid}`
+      const dst = join(realMfsPath('/files'), name)
```

Comment:
Found a bug: relative path within MFS (Files) is not respected. 

If I am in a subdirectory `/a/b` and I try to import `c.car` then the CAR data is added at `/c` rather than `/a/b/c`